### ISSUE TITLE
Error after dfx start with version 0.5.0

### ISSUE THREAD

**Issue by taariq** [enzoh/qr#2]
Looks like main had a problem with the client URL?

```
⠁ Starting up the client...
binding to: V6([::1]:8000)
client: "http://localhost:8080/api"
⠚ Pinging the Internet Computer client...
Feb 24 19:29:07.316 INFO Dfinity Replica Started
Feb 24 19:29:07.317 INFO Existing node ID: 4061159069099793215
Feb 24 19:29:07.331 INFO Using path 'qr/.dfx/state' to manage local state, StateManager: 1
⠂ Pinging the Internet Computer client...
thread 'main' panicked at 'Invalid client URL.: EmptyHost', src/libcore/result.rs:1165:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

**Comment by enzoh** [comment]
Thanks! Let me redirect this to the wider SDK team as it's clearly an issue with `dfx` and not this library. Can you re-run with `RUST_BACKTRACE=1 dfx start` for more information?

**Comment by taariq** [comment]
Yup. Correct.  It's something with dfx. Here's the dump. 

$ RUST_BACKTRACE=1 dfx start
⠁ Starting up the client...
binding to: V6([::1]:8000)
client: "http://localhost:8080/api"
thread 'main' panicked at 'error binding to 0.0.0.0:9091: error creating server listener: Address already in use (os error 48)', /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-crates-io/hyper-0.12.33/src/server/mod.rs:116:17
⠂ Pinging the Internet Computer client...
   0: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
   1: core::fmt::write
   2: std::io::Write::write_fmt
   3: std::panicking::default_hook::{{closure}}
   4: std::panicking::default_hook
   5: std::panicking::rust_panic_with_hook
   6: std::panicking::continue_panic_fmt
   7: std::panicking::begin_panic_fmt
   8: hyper::server::Server<hyper::server::tcp::AddrIncoming,()>::bind
   9: ic_metrics::MetricsRuntimeImpl::new
  10: nodemanager::main
  11: std::rt::lang_start::{{closure}}
  12: std::panicking::try::do_call
  13: __rust_maybe_catch_panic
  14: std::rt::lang_start_internal
  15: main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
thread 'main' panicked at 'error binding to 0.0.0.0:9091: error creating server listener: Address already in use (os error 48)', /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-crates-io/hyper-0.12.33/src/server/mod.rs:116:17
⠂ Pinging the Internet Computer client...
   0: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
   1: core::fmt::write
   2: std::io::Write::write_fmt
   3: std::panicking::default_hook::{{closure}}
   4: std::panicking::default_hook
   5: std::panicking::rust_panic_with_hook
   6: std::panicking::continue_panic_fmt
   7: std::panicking::begin_panic_fmt
   8: hyper::server::Server<hyper::server::tcp::AddrIncoming,()>::bind
   9: ic_metrics::MetricsRuntimeImpl::new
  10: nodemanager::main
  11: std::rt::lang_start::{{closure}}
  12: std::panicking::try::do_call
  13: __rust_maybe_catch_panic
  14: std::rt::lang_start_internal
  15: main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
thread 'main' panicked at 'Invalid client URL.: EmptyHost', src/libcore/result.rs:1165:5
stack backtrace:
thread 'main' panicked at 'error binding to 0.0.0.0:9091: error creating server listener: Address already in use (os error 48)', /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-crates-io/hyper-0.12.33/src/server/mod.rs:116:17
stack backtrace:
   0: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
   1: core::fmt::write
   2: std::io::Write::write_fmt
   3: std::panicking::default_hook::{{closure}}
   0: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
    4: std::panicking::default_hook
  1: core::fmt::write
   2: std::io::Write::write_fmt
   5: std::panicking::rust_panic_with_hook
   3: std::panicking::default_hook::{{closure}}
   4: std::panicking::default_hook
   6: std::panicking::continue_panic_fmt
   5: std::panicking::rust_panic_with_hook
   6: std::panicking::continue_panic_fmt
   7: rust_begin_unwind
   7: std::panicking::begin_panic_fmt
   8: hyper::server::Server<hyper::server::tcp::AddrIncoming,()>::bind
   8: core::panicking::panic_fmt
  9: ic_metrics::MetricsRuntimeImpl::new
  10: nodemanager::main
  11: std::rt::lang_start::{{closure}}
   9: core::result::unwrap_failed
  12: std::panicking::try::do_call
  13: __rust_maybe_catch_panic
  14: std::rt::lang_start_internal
  15: main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
  10: dfx::lib::api_client::Client::new
  11: dfx::commands::start::ping_and_wait
  12: dfx::commands::start::exec
  13: dfx::main
  14: std::rt::lang_start::{{closure}}
  15: std::panicking::try::do_call
  16: __rust_maybe_catch_panic
  17: std::rt::lang_start_internal
  18: main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
taariq$ thread 'main' panicked at 'error binding to 0.0.0.0:9091: error creating server listener: Address already in use (os error 48)', /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-crates-io/hyper-0.12.33/src/server/mod.rs:116:17
stack backtrace:
   0: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
   1: core::fmt::write
   2: std::io::Write::write_fmt
   3: std::panicking::default_hook::{{closure}}
   4: std::panicking::default_hook
   5: std::panicking::rust_panic_with_hook
   6: std::panicking::continue_panic_fmt
   7: std::panicking::begin_panic_fmt
   8: hyper::server::Server<hyper::server::tcp::AddrIncoming,()>::bind
   9: ic_metrics::MetricsRuntimeImpl::new
  10: nodemanager::main
  11: std::rt::lang_start::{{closure}}
  12: std::panicking::try::do_call
  13: __rust_maybe_catch_panic
  14: std::rt::lang_start_internal
  15: main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

**Comment by enzoh** [comment]
Another nodemanager issue.. That got fixed in 0.5.2, which has been released, but is not yet public. try running `for proc in nodemanager replica; do kill -15 $(ps aux | grep $proc | grep -v grep | awk -F ' ' '{print $2}'); done`. Then re-run `dfx start`

**Comment by taariq** [comment]
Copy that. the kill command worked. Now, getting the original error regarding the client URL being empty.  See new trace below.

$ RUST_BACKTRACE=1 dfx start
⠁ Starting up the client...
binding to: V6([::1]:8000)
client: "http://localhost:8080/api"
⠙ Pinging the Internet Computer client...
Feb 24 22:43:32.313 INFO Dfinity Replica Started
Feb 24 22:43:32.313 INFO Existing node ID: 4061159069099793215
Feb 24 22:43:32.314 INFO Using path '/qr/.dfx/state' to manage local state, StateManager: 1
⠂ Pinging the Internet Computer client...
thread 'main' panicked at 'Invalid client URL.: EmptyHost', src/libcore/result.rs:1165:5
⠒ Pinging the Internet Computer client...
   0: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
   1: core::fmt::write
   2: std::io::Write::write_fmt
   3: std::panicking::default_hook::{{closure}}
   4: std::panicking::default_hook
   5: std::panicking::rust_panic_with_hook
   6: std::panicking::continue_panic_fmt
   7: rust_begin_unwind
   8: core::panicking::panic_fmt
   9: core::result::unwrap_failed
  10: dfx::lib::api_client::Client::new
  11: dfx::commands::start::ping_and_wait
  12: dfx::commands::start::exec
  13: dfx::main
  14: std::rt::lang_start::{{closure}}
  15: std::panicking::try::do_call
  16: __rust_maybe_catch_panic
  17: std::rt::lang_start_internal
  18: main

**Comment by enzoh** [comment]
Let's avoid running `dfx start` all together for a moment, since it really just runs the `replica`. kill any instance of the nodemanager and replica as you did above, and then just run `$HOME/.cache/dfinity/versions/0.5.0/replica`. This should start the replica independent of dfx. Then open your web browser and checkout `http://localhost:8080`. If you see something that says "Method not supported", then the replica is running correctly and the issue is on our side. If you cannot connect, then the issue could be a firewall setting on your computer. It sounds very similar to the issue Raul had last week.

**Comment by taariq** [comment]
Yep. Got **Unsupported Method** at localhost. It works. Good to know the replica works on my machine. Very cool!

**Comment by taariq** [comment]
Issue resolved with upgrade to 0.5.2
